File: api/app/api/compliance.py
"""AI-systems register read API — AIC-1 (fork, ADR-F057/F018/F019).

Read-only endpoints over the AI Compliance module's **deployment-global** register
of the organisation's AI systems. The register is the company's standing record
(LQ.AI is single-tenant; ADR-F019), so it is intentionally **shared across the
firm's users**: the cross-user→404 rule that protects private matters does NOT
apply here. The gate is just "active (authenticated) firm user", enforced at the
``include_router`` site (``dependencies=_active``); a 404 means a genuinely missing
record id, not an existence-hiding authz refusal.

Writes are NOT here — the AI Compliance Deep Agent is the only writer, through the
guarded, code-validated tools (``app.agents.compliance_tools``); the user reads and
owns ("system proposes, user owns", ADR-0013 D4). The risk classification is a
legal determination owned by the deterministic engine (AIC-2, ADR-F057), not the
model — this register carries only the FACTS that feed it.
"""

from __future__ import annotations

import uuid
from typing import Annotated

from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.db.session import get_db
from app.models.compliance import AiSystem
from app.schemas.compliance import AiSystemRead

router = APIRouter(prefix="/compliance", tags=["ai-compliance"])

_Db = Annotated[AsyncSession, Depends(get_db)]

_INCLUDE_RETIRED = Query(
    False, description="Include soft-retired rows; default false (live register only)."
)


@router.get("/ai-systems", response_model=list[AiSystemRead])
async def list_ai_systems(db: _Db, include_retired: bool = _INCLUDE_RETIRED) -> list[AiSystem]:
    """The company AI-systems register — all systems (oldest first).

    Retired systems are hidden unless ``include_retired=true``.
    """
    stmt = select(AiSystem).order_by(AiSystem.created_at.asc(), AiSystem.name.asc())
    if not include_retired:
        stmt = stmt.where(AiSystem.retired_at.is_(None))
    rows = (await db.execute(stmt)).scalars().all()
    return list(rows)


@router.get("/ai-systems/{ai_system_id}", response_model=AiSystemRead)
async def get_ai_system(ai_system_id: uuid.UUID, db: _Db) -> AiSystem:
    """One AI system in the register (resolves even if retired, for deep-link/audit)."""
    row = (
        await db.execute(select(AiSystem).where(AiSystem.id == ai_system_id))
    ).scalar_one_or_none()
    if row is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="AI system not found")
    return row

File: web/src/lib/lq-ai/api/compliance.ts
/**
 * /api/v1/compliance — the deployment-global AI-systems register (read-only) — AIC-1.
 *
 * The AI Compliance module's register of the organisation's AI systems under the
 * EU AI Act (Regulation (EU) 2024/1689). The register is the company's standing
 * record (LQ.AI is single-tenant; ADR-F019), so these read endpoints are global —
 * no project/matter id in the path. The AI Compliance Deep Agent is the only
 * writer (guarded, code-validated tools); the UI reads. The register carries FACTS
 * ONLY — the risk tier is a legal determination owned by the classification engine
 * (AIC-2, ADR-F057), not a stored field.
 *
 * Wire shapes mirror api/app/schemas/compliance.py (the Read DTO).
 */
import { apiRequest } from './client';

export type LifecycleStatus = 'in_development' | 'in_service' | 'decommissioned';
export type DevelopmentOrigin = 'in_house' | 'third_party' | 'hybrid';

export interface AiSystemRead {
	id: string;
	name: string;
	intended_purpose: string;
	lifecycle_status: LifecycleStatus;
	development_origin: DevelopmentOrigin;
	is_gpai: boolean;
	gpai_systemic: boolean;
	notes: string | null;
	created_at: string;
	updated_at: string;
	// Soft-retire: null = live (default reads hide retired rows).
	retired_at: string | null;
}

export function listAiSystems(): Promise<AiSystemRead[]> {
	return apiRequest<AiSystemRead[]>('/compliance/ai-systems');
}

export function getAiSystem(id: string): Promise<AiSystemRead> {
	return apiRequest<AiSystemRead>(`/compliance/ai-systems/${encodeURIComponent(id)}`);
}
